payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardDetailsElement.kt

```diff
@@ -73,6 +74,19 @@ internal class CardDetailsElement(
                     IdentifierSpec.CardBrand to FormFieldEntry(it.code, true)
                 }
             )
+            if (isEligibleForCardBrandChoice) {
+                add(
+                    controller.numberElement.controller.cardBrandFlow.map { brand ->
+                        IdentifierSpec.PreferredCardBrand to FormFieldEntry(
+                            value = when (brand) {
+                                CardBrand.Unknown -> null
+                                else -> brand.code
+                            },
```

Comment:
Nit: We can make this a bit more succinct.

```suggestion
                            value = brand.code.takeUnless { brand == CardBrand.Unknown },
```